### ISSUE TITLE
Consider using `credentials` instead of `keyfile`

### ISSUE THREAD

**Issue by blowmage** [GoogleCloudPlatform/google-cloud-ruby#1802]
Now that [`Google::Auth::Credentials`](http://www.rubydoc.info/gems/googleauth/0.6.1/Google/Auth/Credentials) exists and we are going to be replacing our internal `Google::Cloud::Credentials` class with it, should we consider deprecating the `keyfile` argument name and use `credentials` instead? We already allow `keyfile` to be overloaded with a Hash the JSON file contents, or a `Signet::OAuth2::Client` object. Thoughts?

That would change our example code to:

```ruby
require "google/cloud/storage"

storage = Google::Cloud::Storage.new(
  project: "my-todo-project",
  credentials: "/path/to/keyfile.json"
)
```

Currently our example code looks like this:

```ruby
require "google/cloud/storage"

storage = Google::Cloud::Storage.new(
  project: "my-todo-project",
  keyfile: "/path/to/keyfile.json"
)
```

**Comment by quartzmo** [comment]
SGTM to make the addition of `credentials` and update all docs to use `credentials`, but I think we should continue to accept `keyfile` indefinitely as an alias. Why deprecate it?

**Comment by dazuma** [comment]
@quartzmo agreed

**Comment by frankyn** [comment]
+1 to @quartzmo

**Comment by blowmage** [comment]
Deprecated does not mean removed. But if we rename I think we should change the guidance offered in guides and code examples to use `credentials`, and not even mention `keyfile`.

**Comment by blowmage** [comment]
Should we also support this in environmental variable names? So we would consider `LOGGING_CREDENTIALS` as well as `LOGGING_KEYFILE`?

**Comment by blowmage** [comment]
Implemented in #1824.
